File: 0371-sum-of-two-integers/0371-sum-of-two-integers.java
class Solution {
    public int getSum(int a, int b) {
        while (b != 0) {
            // Calculate carry
            int carry = (a & b) << 1;
            
            // Calculate sum without considering carry
            a = a ^ b;
            
            // Update b to carry
            b = carry;
        }
        return a;
    }
}

File: 0371-sum-of-two-integers/0371-sum-of-two-integers.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        mask = 0xFFFFFFFF 
        maxInt = 2**31 - 1 
        while b != 0:
            s = (a ^ b) & mask 
            c = (a & b) & mask 
            a = s 
            b = c << 1
        return a if a <= maxInt else ~(a ^ mask)